### PR TITLE
Add Agent Node Configuration

### LINKED ISSUE #23: Setup Agent node config
### Is your feature request related to a problem? Please describe

Agent node config and keys setup is done manually currently which needs to be automated. 

### Describe the solution you'd like

Secrets Management: The setup should be automated and done using the `CI-Config-Stack` as a part of Jenkins deployment via cdk
Agent Node configuration: TBD

- [ ] ~Set up KMS key that encrypts all secrets~
- [x] Set up the secrets required by agent nodes to communicate with main node
- [x] Set up Groovy scripts that describes the agent node configuration
- [x] Set up the process to execute the groovy scripts (mostly using jenkins-cli)

### PR DESCRIPTION
**Pull request by gaiksaya**
Signed-off-by: Sayali Gaikawad <gaiksaya@amazon.com>

### Description
Adds agent node configuration.
The code currently adds 2 agent nodes configurations AL2-X64 and AL2-ARM64 with 100gb EBS volume. 
We can add windows, macos as an enhancement later. The number to executors (job run per instance) is 2 as of now. Let's discuss what we need to keep it as. 
 
### Issues Resolved
#23 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: lib/ci-stack.ts [hunk 1]
```diff
@@ -80,6 +90,33 @@ export class CIStack extends Stack {
       oidcCredArn: importedOidcConfigValuesSecretBucketValue.toString(),
       useSsl,
       runWithOidc,
+    },
+    {
+      region: this.region.toString(),
+      agent_node_security_group: securityGroups.agentNodeSG.securityGroupId.toString(),
+      subnet_id: vpc.publicSubnets[0].subnetId.toString(),
+    },
+    {
```
Comment (peternied) [inline lib/ci-stack.ts:99]: Lets move these into the `AgentNodeConfig` file, otherwise this `Stack` contains a whole lot of little detail that makes it hard to understand the pieces that are going into the stack.  It could look something like:

```
export class AgentNodeConfig {
   static readonly AGENT_CONFIGS: CloudAgentConfig[] = [{
      ec2_cloud_name: 'AL2-X64',
      ...
   }]
}
```

This way when we add update the configuration only one clearly organized place is updated.

File: lib/ci-stack.ts [hunk 2]
```diff
@@ -80,6 +90,33 @@ export class CIStack extends Stack {
       oidcCredArn: importedOidcConfigValuesSecretBucketValue.toString(),
       useSsl,
       runWithOidc,
+    },
+    {
+      region: this.region.toString(),
```
Comment (peternied) [inline lib/ci-stack.ts:95]: region does not need to be passed in, it can be extracted from `stack.region`

File: lib/compute/jenkins-main-node.ts [hunk 3]
```diff
@@ -329,6 +349,19 @@ export class JenkinsMainNode {
       // Therefore, sleep 60 seconds to wait for plugins to install and jenkins to start which is required for the next step
       InitCommand.shellCommand('sleep 60'),
 
+      // Create groovy script that holds the agent Node config for EC2 plugin ref:https://gist.github.com/vrivellino/97954495938e38421ba4504049fd44ea
+      agentNode.asInitFile('/agent-node-al2-x64.groovy', agentNodeProps, al2x64AgentNodeConfig),
```
Comment (peternied) [inline lib/compute/jenkins-main-node.ts:353]: Can we pass in an `AgentNode` object with a method that produces these InitElement and adds them to the end of the list?  It would reduce the amount of argument passing by a wide margin and reduce coupling

line 137:
```
CloudFormationInit.fromElements(...JenkinsMainNode.configElements(...),
                                ...props.cloudAgent.configElements())
```

File: lib/security/ci-security-groups.ts [hunk 4]
```diff
@@ -26,7 +26,13 @@ export class JenkinsSecurityGroups {
       vpc,
       description: 'Main node of Jenkins',
     });
+
+    this.agentNodeSG = new SecurityGroup(stack, 'AgentNodeSG', {
```
Comment (peternied) [inline lib/security/ci-security-groups.ts:30]: move the declaration and the additional rules after the main agent nodes, so it reads more like 

* External Access SG / Rules 
* MainNode SG / Rules
* AgentNode SG / Rules

File: lib/compute/jenkins-main-node.ts [hunk 5]
```diff
@@ -34,6 +35,14 @@ export interface JenkinsMainNodeProps extends HttpConfigProps, OidcFederateProps
   readonly sg: SecurityGroup;
 }
 
+export interface AgentNodeProps{
+  readonly region : string;
+  // eslint-disable-next-line camelcase
```
Comment (peternied) [inline lib/compute/jenkins-main-node.ts:40]: This interface should follow our normal rules since it isn't `toJSON`ed

File: lib/compute/agent-nodes.ts [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+      assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
+    });
+    const AgentNodeInstanceProfile = new CfnInstanceProfile(stack, 'JenkinsAgentNodeInstanceProfile', { roles: [AgentNodeRole.roleName] });
+    this.AgentNodeInstanceProfileArn = AgentNodeInstanceProfile.attrArn.toString();
+    this.SSHEC2KeySecretId = Fn.join('/', ['ec2-ssh-key', key.keyPairName.toString(), 'private']);
+  }
+
+  public asInitFile(filePath: string, stackConfig: AgentNodeProps, config: AgentNodeConfig, options?: InitFileOptions): InitFile {
+    return InitFile.fromString(filePath,
+      `import hudson.model.*
```
Comment (peternied) [inline lib/compute/agent-nodes.ts:52]: I know this is a big hunk of groovy in a string; it would be easier to read if we carefully indented and aligned columns just like it was our typescript code.

File: lib/compute/agent-nodes.ts [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    import jenkins.model.*
+    import hudson.plugins.ec2.*
+    import com.amazonaws.services.ec2.model.InstanceType
+
+    def instance = Jenkins.getInstance()
+    def init_script = 'sudo mkdir -p /var/jenkins/ && sudo chown -R ec2-user:ec2-user /var/jenkins \
+    && sudo yum install -y java-1.8.0-openjdk cmake python3 python3-pip && sudo yum groupinstall -y "Development Tools" \
```
Comment (peternied) [inline lib/compute/agent-nodes.ts:59]: Since this are line continuations please indent, occurs in a number of places

File: lib/compute/agent-nodes.ts [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+    && sudo ln -sfn \`which pip3\` /usr/bin/pip && pip3 install pipenv && sudo ln -s ~/.local/bin/pipenv /usr/local/bin'
+
+// variables
```
Comment (peternied) [inline lib/compute/agent-nodes.ts:62]: Align the comment with the definition, and it isn't very clear to me based on naming the difference between this grouping and the one below it.

File: lib/compute/agent-nodes.ts [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+// info imported the stack
+    def region = "${stackConfig.region}"
+    def security_group = "${stackConfig.agent_node_security_group}"
+    def subnet_id = "${stackConfig.subnet_id}"
+    def instance_profile_arn = "${this.AgentNodeInstanceProfileArn}"
+    def sshKeysASMSecretId = "${this.SSHEC2KeySecretId}"
+    def ec2_tags = [new EC2Tag('Name', 'jenkins-agent-node')]
+
+    def agent_node_template = new SlaveTemplate(
```
Comment (peternied) [inline lib/compute/agent-nodes.ts:78]: This code is tightly coupled to `SlaveTemplate` could you link where this interface is described so future maintainers can use it for follow up?

File: lib/compute/agent-nodes.ts [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+      region,
+      // String privateKey
+      null,
+      // String sshKeysCredentialsId
+      sshKeysASMSecretId,
+      // String instanceCapStr
+      "10",
+      // List<? extends SlaveTemplate> templates
+      [agent_node_template],
+      // String roleArn
```
Comment (peternied) [inline lib/compute/agent-nodes.ts:176]: Is this the role of the running agent node? I think we should have one so we can use SSM to ssh into the host just like the [main node](https://github.com/opensearch-project/opensearch-ci/blob/86aea2e78a29cd02e1df77031a1b0b57e09f2dd9/lib/compute/jenkins-main-node.ts#L70) for troubleshooting.

File: lib/compute/agent-nodes.ts (hunk 9 above)
Comment (peternied) [reply lib/compute/agent-nodes.ts:78]: Note; also applies to the AmazonEC2Cloud constructor below.

File: lib/compute/agent-nodes.ts [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+ */
+
+import { CfnInstanceProfile, Role, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { Fn, Stack, Tags } from '@aws-cdk/core';
+import { KeyPair } from 'cdk-ec2-key-pair';
+import { InitFile, InitFileOptions } from '@aws-cdk/aws-ec2';
+import { AgentNodeProps } from './jenkins-main-node';
+
+export interface AgentNodeConfig{
+  // eslint-disable-next-line camelcase
```
Comment (peternied) [inline lib/compute/agent-nodes.ts:16]: This interfaces style should follow normal conventions

File: lib/compute/agent-nodes.ts (hunk 10 above)
Comment (peternied) [reply lib/compute/agent-nodes.ts:176]: - it looks like the instance profile has a role, maybe a comment would be useful instead

File: lib/compute/agent-nodes.ts (hunk 7 above)
Comment (peternied) [inline lib/compute/agent-nodes.ts:59]: Please break apart into one operation per line for ease of reading

Comment (peternied) [reply lib/compute/agent-nodes.ts:59]: Actually, could you declare this out side of the groovy and import it?

File: lib/ci-stack.ts (hunk 2 above)
Comment (gaiksaya) [reply lib/ci-stack.ts:95]: The method is not a part of the constructor so need to pass that as a parameter https://github.com/opensearch-project/opensearch-ci/pull/39/files#diff-90fd20658ffdeb68fbac6a146498601102a44c741138481202ff494189e69c30R50
Should I pass stack as a param then and then extract region from it?

Comment (peternied) [reply lib/ci-stack.ts:95]: Sure, but do that at the last possible minute in the code so you aren't passing that parameter around when it is unused.

File: lib/compute/agent-nodes.ts (hunk 6 above)
Comment (abhinavGupta16) [reply lib/compute/agent-nodes.ts:52]: Alternatively, would it be better to maybe write the groovy code as a groovy file and import it? 
Values can then later be replaced as variables from the props. This would make the groovy code more readable and IDEs would also be able to parse it to verify for errors etc

**Comment by gaiksaya** [comment]
> @abhinavGupta16 Alternatively, would it be better to maybe write the groovy code as a groovy file and import it?
Values can then later be replaced as variables from the props. This would make the groovy code more readable and IDEs would also be able to parse it to verify for errors etc<
I did think about that, about using [InitFile.fromAsset](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.InitFile.html#static-fromwbrassettargetfilename-path-options) but how would we change the values once the file is created in the ec2? These files contain dynamic values like SG, subnet ids, etc. How should we replace that?

**Comment by peternied** [comment]
> > @abhinavGupta16 Alternatively, would it be better to maybe write the groovy code as a groovy file and import it?
> > Values can then later be replaced as variables from the props. This would make the groovy code more readable and IDEs would also be able to parse it to verify for errors etc
> 
> I did think about that, about using [InitFile.fromAsset](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.InitFile.html#static-fromwbrassettargetfilename-path-options) but how would we change the values once the file is created in the ec2 or even before it? These files contain dynamic values like SG, subnet ids, etc. How should we replace that? Will look into it if we can use `Cfn:Fn:sub` like I used it [here](https://github.com/opensearch-project/opensearch-infra/blob/main/opensearch-cluster/cdk/multi-node/multi_node_cdk/infra.py#L167)

I would recommend not spending too many cycles here, while not ideal there is minimal functionality in that groovy code that is cannot be validated other than seeing a fresh main node spin up.  If we have more ideas lets consider making issues for them to resolve over time.

**Comment by abhinavGupta16** [comment]
> > > @abhinavGupta16 Alternatively, would it be better to maybe write the groovy code as a groovy file and import it?
> > > Values can then later be replaced as variables from the props. This would make the groovy code more readable and IDEs would also be able to parse it to verify for errors etc
> > 
> > 
> > I did think about that, about using [InitFile.fromAsset](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.InitFile.html#static-fromwbrassettargetfilename-path-options) but how would we change the values once the file is created in the ec2 or even before it? These files contain dynamic values like SG, subnet ids, etc. How should we replace that? Will look into it if we can use `Cfn:Fn:sub` like I used it [here](https://github.com/opensearch-project/opensearch-infra/blob/main/opensearch-cluster/cdk/multi-node/multi_node_cdk/infra.py#L167)
> 
> I would recommend not spending too many cycles here, while not ideal there is minimal functionality in that groovy code that is can be validated other than starting up a fresh main node and validating it. If we have more ideas lets consider making issues for them to resolve seperately.

@peternied - We can definitely create an issue and do it the separate cycle. 
@gaiksaya - A few says I can think of is taking variables as a parameter to the script and running it. Alternatively, we can read the entire file as a string in the typescript code and replace the variable values. Another, not so nice method could be saving the file on EC2 and replace the values on the file using cmd.

File: lib/ci-stack.ts [hunk 12]
```diff
@@ -6,7 +6,17 @@
  * compatible open source license.
  */
 
-import { FlowLogDestination, FlowLogTrafficType, Vpc } from '@aws-cdk/aws-ec2';
+import * as ec2 from '@aws-cdk/aws-ec2';
```
[the remark this answers is not in the document]
Comment (gaiksaya) [reply lib/ci-stack.ts]: Corrected

File: lib/ci-stack.ts (hunk 1 above)
Comment (gaiksaya) [reply lib/ci-stack.ts]: Done

File: lib/ci-stack.ts (hunk 2 above)
Comment (gaiksaya) [reply lib/ci-stack.ts]: Done

File: lib/security/ci-security-groups.ts (hunk 4 above)
Comment (gaiksaya) [reply lib/security/ci-security-groups.ts:33]: Done

Comment (gaiksaya) [reply lib/security/ci-security-groups.ts:33]: Done

File: lib/compute/agent-nodes.ts (hunk 9 above)
Comment (gaiksaya) [reply lib/compute/agent-nodes.ts:76]: Added

File: lib/compute/agent-nodes.ts (hunk 10 above)
Comment (gaiksaya) [reply lib/compute/agent-nodes.ts]: Added

File: lib/compute/agent-nodes.ts (hunk 7 above)
Comment (gaiksaya) [reply lib/compute/agent-nodes.ts]: Done

File: lib/compute/agent-nodes.ts [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+  ami_id : string;
```
[the remark this answers is not in the document]
Comment (gaiksaya) [reply lib/compute/agent-nodes.ts]: Done

File: package-lock.json [hunk 14]
```diff
@@ -1,13190 +1,8 @@
 {
   "name": "opensearch-ci-stack",
   "version": "0.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (peternied) [inline package-lock.json:4]: Can you revert this change?

**Review by peternied — COMMENTED** [review]
Looking really good, should it move out of draft or did you have more changes in mind?

**Comment by gaiksaya** [comment]
> Looking really good, should it move out of draft or did you have more changes in mind?

All tests are broken. And I missed one of your comments which will really clean things up

>Can we pass in an AgentNode object with a method that produces these InitElement and adds them to the end of the list? It would reduce the amount of argument passing by a wide margin and reduce coupling
line 137:
CloudFormationInit.fromElements(...JenkinsMainNode.configElements(...),
                                ...props.cloudAgent.configElements())


So still working on it.

**Comment by peternied** [comment]
> > > > @abhinavGupta16 Alternatively, would it be better to maybe write the groovy code as a groovy file and import it?
> > > > Values can then later be replaced as variables from the props. This would make the groovy code more readable and IDEs would also be able to parse it to verify for errors etc
> > > 
> > > 
> > > I did think about that, about using [InitFile.fromAsset](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.InitFile.html#static-fromwbrassettargetfilename-path-options) but how would we change the values once the file is created in the ec2 or even before it? These files contain dynamic values like SG, subnet ids, etc. How should we replace that? Will look into it if we can use `Cfn:Fn:sub` like I used it [here](https://github.com/opensearch-project/opensearch-infra/blob/main/opensearch-cluster/cdk/multi-node/multi_node_cdk/infra.py#L167)
> > 
> > 
> > I would recommend not spending too many cycles here, while not ideal there is minimal functionality in that groovy code that is can be validated other than starting up a fresh main node and validating it. If we have more ideas lets consider making issues for them to resolve seperately.
> 
> @peternied - We can definitely create an issue and do it the separate cycle. The reason I ask is because this would make our code more readable and maintainable.

Making the largest thread I can! For your consideration, readability and maintainability are features that have a time cost, sometimes we just need action (consider the effort and follow up tasks around docker generation) other times we need careful construction for long term payoff (consider our internal deployment system, its hardly even started and we are still planning it!) it is always important to ask how likely the time will return dividends.

Supporting groovy code in this project, as is, doesn't seem very fruitful to me; however, if we had a use case like DB is addressing in https://github.com/opensearch-project/opensearch-build/pull/924 a systemic problem that has hampered us many times.  We should always make sure we have a goal in mind and are aware of the constraints we place on what we build.

File: lib/compute/jenkins-main-node.ts (hunk 3 above)
Comment (gaiksaya) [reply lib/compute/jenkins-main-node.ts]: Done

**Comment by gaiksaya** [comment]
Opened up issue for loading groovy scripts from files: https://github.com/opensearch-project/opensearch-ci/issues/44

I have not added tests for agent-nodes specifically as methods are not static and need to instatiate the class which is causing me to run into 
```
 There is already a Construct with name 'AgentNode-KeyPair' in CIStack [MyTestStack]
```

Not sure if tests are required for agent-nodes and they itself are dynamic.

File: package-lock.json (hunk 14 above)
Comment (gaiksaya) [reply package-lock.json:4]: This should be 1 or 2?

File: README.md [hunk 15]
```diff
@@ -44,6 +44,7 @@ OpenSearch Continuous Integration is an open source CI system for OpenSearch and
 6. Log onto the AWS Console of the account, navigate to [cloud watch](https://console.aws.amazon.com/cloudwatch/home), open log groups, looking for `JenkinsMainNode/var/log/jenkins/jenkins.log`
 7. Search the logs for `Jenkins initial setup is required. An admin user has been created and a password generated.` After that entry the password for the jenkins instance will be in the cloudwatch logs.
 8. Go to the `CI-Dev.JenkinsExternalLoadBalancerDns` url returned by CDK output to access the jenkins host.
+9. While destroying the stack make sure you delete the agent nodes manually (via jenkins UI or AWS console) so that shared resources (like vpc, security groups, etc) can be deleted.
```
Comment (peternied) [inline README.md:47]: Switch to -> `If you destroy the stack,` I never destroy my stacks, and I wouldn't recommend it to most developers unless you are doing specific setup where you need a fresh environment.

File: lib/ci-stack.ts [hunk 16]
```diff
@@ -80,7 +85,13 @@ export class CIStack extends Stack {
       oidcCredArn: importedOidcConfigValuesSecretBucketValue.toString(),
       useSsl,
       runWithOidc,
-    });
+    },
+    {
+      agentNodeSecurityGroup: securityGroups.agentNodeSG.securityGroupId.toString(),
```
Comment (peternied) [inline lib/ci-stack.ts:90]: Don't toString these, you can pass around the security group and have it tostring before its used if needed

File: lib/ci-stack.ts [hunk 17]
```diff
@@ -80,7 +85,13 @@ export class CIStack extends Stack {
...
+      subnetId: vpc.publicSubnets[0].subnetId.toString(),
+    },
+    agentNodesConfig.AL2_X64,
```
Comment (peternied) [inline lib/ci-stack.ts:93]: Can you have the main node accept an agentNodesConfig and then it extracts what it needs?

File: lib/ci-stack.ts (hunk 16 above)
Comment (peternied) [inline lib/ci-stack.ts:90]: Seems like the agent configuration should be passed into the CloudAgentNodeConfig object instead of into the jenkins main node.

File: lib/compute/agent-node-config.ts [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+  readonly AL2_ARM64: AgentNodeConfig;
+
+  constructor(stack: Stack) {
+    this.AL2_X64 = {
+      ec2CloudName: 'AL2-X64',
+      instanceType: InstanceType.of(InstanceClass.C5, InstanceSize.XLARGE4).toString(),
+      workerLabelString: 'AL2-X64',
+      numberOfExecutors: '2',
+      remoteUser: 'ec2-user',
```
Comment (peternied) [inline lib/compute/agent-node-config.ts:26]: Any reason we can't hardcode this value?

File: lib/compute/agent-node-config.ts [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+      amiId: MachineImage.latestAmazonLinux({
+        generation: AmazonLinuxGeneration.AMAZON_LINUX_2,
+        cpuType: AmazonLinuxCpuType.X86_64,
+      }).getImage(stack).imageId.toString(),
+    };
+    this.AL2_ARM64 = {
+      ec2CloudName: 'AL2-ARM64',
+      instanceType: InstanceType.of(InstanceClass.C6G, InstanceSize.XLARGE4).toString(),
+      workerLabelString: 'AL2-ARM64',
+      numberOfExecutors: '2',
```
Comment (peternied) [inline lib/compute/agent-node-config.ts:36]: Can we hardcode this value too?

File: lib/compute/agent-nodes.ts [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+      '',
+      // String roleSessionName
+      ''
+    )
+
+    instance.clouds.add(new_cloud)
+    instance.save()`);
+  }
+
+  public configElements(stackRegion: string, agentNodeProps: AgentNodeProps, al2x64AgentNodeConfig: AgentNodeConfig,
```
Comment (peternied) [inline lib/compute/agent-nodes.ts:192]: These props seem like they can be fields of this class rather than being passed around

File: lib/compute/jenkins-main-node.ts [hunk 21]
```diff
@@ -34,6 +35,11 @@ export interface JenkinsMainNodeProps extends HttpConfigProps, OidcFederateProps
   readonly sg: SecurityGroup;
 }
 
+export interface AgentNodeProps{
```
Comment (peternied) [inline lib/compute/jenkins-main-node.ts:38]: Move to the agent node class

**Review by peternied — APPROVED** [review]
Feedback is mostly nitpicking to improve encapsulation, looks pretty good.

File: lib/compute/agent-node-config.ts (hunk 18 above)
Comment (gaiksaya) [reply lib/compute/agent-node-config.ts:26]: I am using same groovy script to install 2 types of agent nodes inorder to avoid duplication. Maybe we can refactor later when we implement #44 . So we can have multiple groovy scripts for each type of agent node

File: lib/compute/agent-node-config.ts (hunk 19 above)
Comment (gaiksaya) [reply lib/compute/agent-node-config.ts:36]: Same as above

File: lib/ci-stack.ts (hunk 17 above)
Comment (gaiksaya) [reply lib/ci-stack.ts]: Done

File: lib/ci-stack.ts (hunk 16 above)
Comment (gaiksaya) [reply lib/ci-stack.ts]: Done

File: lib/compute/jenkins-main-node.ts (hunk 21 above)
Comment (gaiksaya) [reply lib/compute/jenkins-main-node.ts]: I did, Forgot to remove from here. Removed now

File: lib/compute/agent-nodes.ts (hunk 20 above)
Comment (gaiksaya) [reply lib/compute/agent-nodes.ts:192]: Will refactor later is required. Getting this PR in for now.

**Comment by gaiksaya** [comment]
@peternied Are you okay with me merging this PR? Any objection?

**Review by abhinavGupta16 — APPROVED** [review]

**Comment by peternied** [comment]
@gaiksaya I've already signed off you can merge whenever you'd like.

If you were open to changes, I'd recommend reverting the changes to `package-lock.json`

**Comment by gaiksaya** [comment]
> @gaiksaya I've already signed off you can merge whenever you'd like.
> 
> If you were open to changes, I'd recommend reverting the changes to `package-lock.json`

Is it just `lockfileVersion`? or entire package-lock.json?

**Comment by abhinavGupta16** [comment]
> > @gaiksaya I've already signed off you can merge whenever you'd like.
> > If you were open to changes, I'd recommend reverting the changes to `package-lock.json`
> 
> Is it just `lockfileVersion`? or entire package-lock.json?

I don't think you revert the entire file since you have a change in `package.json` which should now reflect in `package-lock.json`

**Comment by gaiksaya** [comment]
> > @gaiksaya I've already signed off you can merge whenever you'd like.
> > If you were open to changes, I'd recommend reverting the changes to `package-lock.json`
> 
> Is it just `lockfileVersion`? or entire package-lock.json?

Yeah since the file is auto generated concerned if I should change anything at all

**Comment by gaiksaya** [comment]
> > > @gaiksaya I've already signed off you can merge whenever you'd like.
> > > If you were open to changes, I'd recommend reverting the changes to `package-lock.json`
> > 
> > 
> > Is it just `lockfileVersion`? or entire package-lock.json?
> 
> Yeah since the file is auto generated , I am concerned if I should change anything at all

Update: Reverted the package-lock.json changes and ran `npm run build` the project is still building. Merging this PR.
Thanks!
